debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlValueConverters.java

```diff
@@ -0,0 +1,232 @@
+        // Handle a few MySQL-specific types based upon how they are handled by the MySQL binlog client ...
+        String typeName = column.typeName().toUpperCase();
+        if (matches(typeName, "YEAR")) {
+            return (data) -> convertYear(column, fieldDefn, data);
+        }
+        if (matches(typeName, "ENUM")) {
+            // Build up the character array based upon the column's type ...
+            String options = extractEnumAndSetOptions(column,false);
+            return (data) -> convertEnum(options, column, fieldDefn, data);
+        }
+        if (matches(typeName, "SET")) {
+            // Build up the character array based upon the column's type ...
+            String options = extractEnumAndSetOptions(column,false);
+            return (data) -> convertSet(options, column, fieldDefn, data);
+        }
+        // Otherwise, let the base class handle it ...
+        return super.converter(column, fieldDefn);
+    }
+
+    /**
+     * Converts a value object for a MySQL {@code YEAR}, which appear in the binlog as an integer though returns from
+     * the MySQL JDBC driver as either a short or a {@link java.sql.Date}.
+     * 
+     * @param column the column definition describing the {@code data} value; never null
+     * @param fieldDefn the field definition; never null
+     * @param data the data object to be converted into a {@link Date Kafka Connect date} type; never null
+     * @return the converted value, or null if the conversion could not be made
+     */
+    @SuppressWarnings("deprecation")
+    protected Object convertYear(Column column, Field fieldDefn, Object data) {
+        if (data == null) return null;
+        if (data instanceof java.time.Year) {
+            // The MySQL binlog always returns a Year object ...
+            return ((java.time.Year) data).getValue();
+        }
+        if (data instanceof java.sql.Date) {
+            // MySQL JDBC driver sometimes returns a Java SQL Date object ...
+            return ((java.sql.Date) data).getYear();
+        }
+        if (data instanceof Number) {
+            // MySQL JDBC driver sometimes returns a short ...
+            return ((Number) data).intValue();
+        }
+        return handleUnknownData(column, fieldDefn, data);
+    }
+
+    /**
+     * Converts a value object for a MySQL {@code ENUM}, which is represented in the binlog events as an integer value containing
+     * the index of the enum option. The MySQL JDBC driver returns a string containing the option,
+     * so this method calculates the same.
+     * 
+     * @param options the characters that appear in the same order as defined in the column; may not be null
+     * @param column the column definition describing the {@code data} value; never null
+     * @param fieldDefn the field definition; never null
+     * @param data the data object to be converted into a {@link Date Kafka Connect date} type; never null
```

Comment:
the javadoc for this is slightly "stale", i.e. this will return a string representation, not a date.